hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -171,9 +172,17 @@ public boolean dropTable(TableIdentifier identifier, boolean purge) {
             false /* throw NoSuchObjectException if the table doesn't exist */);
         return null;
       });
-
       if (purge && lastMetadata != null) {
         CatalogUtil.dropTableData(ops.io(), lastMetadata);
+
+        Boolean shouldDropFullTableFolder = Boolean.parseBoolean(lastMetadata.property(
+            TableProperties.DROP_FULL_TABLE_FOLDER,
+            String.valueOf(TableProperties.DROP_FULL_TABLE_FOLDER_DEFAULT)));
+        if (shouldDropFullTableFolder) {
+          ((HadoopFileIO) fileIO).deletePrefix(lastMetadata.location());
```

Comment:
I this is not something we can build upon.
See:
```
    this.fileIO = fileIOImpl == null ? new HadoopFileIO(conf) : CatalogUtil.loadFileIO(fileIOImpl, properties, conf);
```

I would do something like this:
```
if (fileIO instanceof SupportsPrefixOperations) {
    ((SupportsPrefixOperations) fileIO).deletePrefix(lastMetadata.location());
}
```

Reply:
Indeed, thx!